include/onnxruntime/core/platform/threadpool.h

```diff
@@ -118,27 +118,30 @@ class ThreadPool {
 #else
   using NAME_CHAR_TYPE = char;
 #endif
-  // Constructs a pool that contains "num_threads" threads with specified
-  // "name". env->StartThread() is used to create individual threads with the
-  // given ThreadOptions. If "low_latency_hint" is true the thread pool
+  // Constructs a pool for running with with "degree_of_parallelism" threads with
+  // specified "name". env->StartThread() is used to create individual threads
+  // with the given ThreadOptions. If "low_latency_hint" is true the thread pool
   // implementation may use it as a hint that lower latency is preferred at the
   // cost of higher CPU usage, e.g. by letting one or more idle threads spin
   // wait. Conversely, if the threadpool is used to schedule high-latency
   // operations like I/O the hint should be set to false.
   //
-  // REQUIRES: num_threads > 0
+  // REQUIRES: degree_of_parallelism > 0
   // The allocator parameter is only used for creating a Eigen::ThreadPoolDevice to be used with Eigen Tensor classes.
   ThreadPool(Env* env,
              const ThreadOptions& thread_options,
              const NAME_CHAR_TYPE* name,
-             int num_threads,
+             int degree_of_parallelism,
              bool low_latency_hint);
 
   // Waits until all scheduled work has finished and then destroy the
   // set of threads.
   ~ThreadPool();
 
-  // Schedules fn() for execution in the pool of threads.
+  // Schedules fn() for execution in the pool of threads.  The function may run
+  // synchronously if it cannot be enqueued.  This will occur if the thread pool's
+  // degree-of-parallelism is 1, but it may also occur for implementation-dependent
+  // reasons such as if queues used for buffering work are full.
   void Schedule(std::function<void()> fn);
```

Comment:
One question:
What if DegreeOfParallelism is 4 but the parrallel for function cut the computation job into 8 pieces? How the sub tasks will be scheduled to each of the threads? Will the main thread take one of the 8 or two?

Reply:
For parallel loops, the functions scheduled onto the workers are running the lambda at https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/common/threadpool.cc#L161 (in contrast to directly scheduling work items for specific iterations).  The lambda will claim iterations for the parallel loop until they are all executed, so will rebalance the work across the set of threads participating in the loop.  In this example, if the iterations are all equal size, then each thread should run two unless the loop is so short that loop entry/exit costs dominate.